File: graphql-api/src/main/java/io/prizy/publicapi/graphql/directives/AuthorizedDirective.java
package io.prizy.publicapi.graphql.directives;

import graphql.schema.GraphQLFieldDefinition;
import graphql.schema.idl.SchemaDirectiveWiring;
import graphql.schema.idl.SchemaDirectiveWiringEnvironment;

/**
 * @author Nidhal Dogga
 * @created 6/24/2022 8:54 PM
 */


public class AuthorizedDirective implements SchemaDirectiveWiring {

  public static final String NAME = "authorized";

  @Override
  public GraphQLFieldDefinition onField(SchemaDirectiveWiringEnvironment<GraphQLFieldDefinition> environment) {
    return SchemaDirectiveWiring.super.onField(environment);
  }

}

File: graphql-core/src/main/kotlin/io/prizy/graphql/directives/AuthorizedDirective.kt
package io.prizy.graphql.directives

import com.expediagroup.graphql.generator.annotations.GraphQLDirective
import graphql.introspection.Introspection.DirectiveLocation.FIELD_DEFINITION

/**
 *  @author Nidhal Dogga
 *  @created 11/27/2021 11:54 AM
 *  @credits prizy.io
 *  All rights reserved
 */

@GraphQLDirective(
  name = AuthorizedDirective.NAME,
  locations = [FIELD_DEFINITION]
)
annotation class AuthorizedDirective(
  vararg val roles: String = []
) {

  companion object {
    const val NAME = "authorized"
  }
}
